zcutil/fetch-params.sh

```diff
@@ -69,26 +89,33 @@ Creating params directory. For details about this directory, see:
 $README_PATH
 
 EOF
-fi
-
-mkdir -p "$REGTEST_DIR"
+    fi
 
-fetch_params "$REGTEST_PKEY_URL" "$REGTEST_DIR/$REGTEST_PKEY_NAME"
-fetch_params "$REGTEST_VKEY_URL" "$REGTEST_DIR/$REGTEST_VKEY_NAME"
+    mkdir -p "$REGTEST_DIR"
 
-echo 'Updating testnet3 symlinks to regtest parameters.'
-mkdir -p "$TESTNET3_DIR"
-ln -sf "../regtest/$REGTEST_PKEY_NAME" "$TESTNET3_DIR/$REGTEST_PKEY_NAME"
-ln -sf "../regtest/$REGTEST_VKEY_NAME" "$TESTNET3_DIR/$REGTEST_VKEY_NAME"
+    fetch_params "$REGTEST_PKEY_URL" "$REGTEST_DIR/$REGTEST_PKEY_NAME"
+    fetch_params "$REGTEST_VKEY_URL" "$REGTEST_DIR/$REGTEST_VKEY_NAME"
 
-cd "$PARAMS_DIR"
+    cd "$PARAMS_DIR"
 
-# Now verify their hashes:
-echo 'Verifying parameter file integrity via sha256sum...'
-shasum -a 256 --check <<EOF
+    # Now verify their hashes:
+    echo 'Verifying parameter file integrity via sha256sum...'
+    shasum -a 256 --check <<EOF
 226913bbdc48b70834f8e044d194ddb61c8e15329f67cdc6014f4e5ac11a82ab  regtest/$REGTEST_PKEY_NAME
-226913bbdc48b70834f8e044d194ddb61c8e15329f67cdc6014f4e5ac11a82ab  testnet3/$REGTEST_PKEY_NAME
 4c151c562fce2cdee55ac0a0f8bd9454eb69e6a0db9a8443b58b770ec29b37f5  regtest/$REGTEST_VKEY_NAME
-4c151c562fce2cdee55ac0a0f8bd9454eb69e6a0db9a8443b58b770ec29b37f5  testnet3/$REGTEST_VKEY_NAME
 EOF
+    # Check the exit code of the shasum command:
+    CHECKSUM_RESULT=$?
+    if [ $CHECKSUM_RESULT -eq 0 ]; then
+        echo 'Updating testnet3 symlinks to regtest parameters.'
+        mkdir -p "$TESTNET3_DIR"
+        ln -sf "../regtest/$REGTEST_PKEY_NAME" "$TESTNET3_DIR/$REGTEST_PKEY_NAME"
+        ln -sf "../regtest/$REGTEST_VKEY_NAME" "$TESTNET3_DIR/$REGTEST_VKEY_NAME"
+    else
+       exit 1
```

Comment:
@str4d So, I tried to put an error message on the other side of the conditional, except the funny thing about that is it will never get triggered. Because of `set -e`, the entire script will immediately exit when `shasum` returns a bad exit code, and we'll never get a code path that allows us to display an error. So it's kind of funny/redundant to check the exit code of `shasum` to begin with (but wasn't it one of the audit items?).